### PR TITLE
Issue #6   rawphenotypes hook alter() implementations

### PR DESCRIPTION
**Pull request by laceysanderson**
This pull request should have been created when the changes were submitted for review. See Issue #6

### CODE REVIEW

File: kp_nodes.module [hunk 1]
```diff
@@ -504,3 +504,133 @@ function kp_project_germplasm_list($project = null) {
...
+ *
+ * @return
+ *   A string, email address.
+ *
+ * NOTE: to set this value, run/execute this line in Drupal /devel/php
+ *       variable_set('kp_rawpheno_support_email', 'your-email@email.com')
+ *       otherwise, the module will load email from either the Drupal user 0 
+ *       or the Drupal site wide mail.
+ */
+function kp_nodes_rawpheno_support_email_alter(&$support_email) {
```
Comment (laceysanderson) [inline kp_nodes.module]: This function should only provide @reynoldtan's email. As mentioned in the related pull request the default email should be provided via rawphenotypes.

File: kp_nodes.module [hunk 2]
```diff
@@ -504,3 +504,133 @@ function kp_project_germplasm_list($project = null) {
...
+ * @param $columns
+ *   An array, as set of column headers from a spreadsheet file.
+ *
+ * @return
+ *   An array, the same column headers in the parameter but with the
+ *   specified column(s) removed.
+ */
+
+// When searching for new columns.
+// in rawpheno.upload.form.inc : rawpheno_upload_form_stage_review().
```
Comment (laceysanderson) [inline kp_nodes.module]: Please be more descriptive in your docblocks. Also there should not be any code between the docblock and the function header. The kp_nodes_rawpheno_support_email_alter() docblock was perfect :-) so just keep with that format for All Alter functions while substituting in the correct info (e.g. columns, etc.).

File: kp_nodes.module [hunk 3]
```diff
@@ -504,3 +504,133 @@ function kp_project_germplasm_list($project = null) {
...
+        unset($header[$i]);
+      }
+    }
+  }
+}
+
+// When saving and validating data.
+// in rawpheno.upload.form.inc : rawpheno_validate_excel_file().
+//                               rawpheno_load_spreadsheet().
+function kp_nodes_rawpheno_ignorecols_valsave_alter(&$skip, &$project_name) {
```
Comment (laceysanderson) [inline kp_nodes.module]: Need Docblock

File: kp_nodes.module [hunk 4]
```diff
@@ -504,3 +504,133 @@ function kp_project_germplasm_list($project = null) {
...
+    $t = kp_nodes_rawpheno_ignorecols();
+    $index = array_map('rawpheno_function_delformat', $t);
+
+    if (count($index) > 0) {
+      $skip = $index;
+    }
+  }
+}
+
+// Check existence of column(s) in columns array.
```
Comment (laceysanderson) [inline kp_nodes.module]: Need docblock

File: kp_nodes.module [hunk 5]
```diff
@@ -504,3 +504,133 @@ function kp_project_germplasm_list($project = null) {
...
+
+    if (in_array($c, $ignore_cols)) {
+      $cols_i[] = $i;
+    }
+  }
+
+  return $cols_i;
+}
+
+// Define a set columns to ignore.
```
Comment (laceysanderson) [inline kp_nodes.module]: Need docblock

**Review by laceysanderson — COMMENTED** [review]
The functional code looks good; however, function header doc blocks need to be improved

**Comment by reynoldtan** [comment]
Updated code and docblock documentation. This update requires Issue-#36---Create-a-function-to-fetch-support-email pull request.

**Comment by laceysanderson** [comment]
🎉

**Pull request merged**